packages/circuit-ui/components/Theming/Theming.docs.mdx

```diff
@@ -0,0 +1,168 @@
+    "primary": "darkblue",
+    "onPrimary": "white"
+  },
+  "bg": {
+    // neutrals
+    "default": {
+      "idle": "white",
+      "hover": "lightestgray",
+      "active": "lightergray"
+    },
+    "subtle": {
+      "idle": "lightergray",
+      "hover": "lightgray",
+      "active": "gray"
+    },
+    // roles (example using primary, same for confirm, notify, alert)
+    "primary": {
+      "default": {
+        "idle": "darkblue",
+        "hover": "darkerblue",
+        "active": "darkestblue"
+      }
+      // same for other role bg variants, e.g. subtle
+    }
+  },
+  "border": {} // same as bg
+}
+```
+
+### Rationale
+
+- On the top-level, we split foreground colors and background/border colors, because foreground colors have no variants (except neutrals) and states* (*: see open questions)
+- We split neutral colors and role colors, because neutrals typically have more variants (e.g. default neutral text, subtle neutral text, highlight neutral text—but only a single variant of alert text)
+- We split backgrounds and borders because they vary in some cases (e.g. in the light theme, default bg is white and default border is light gray), and also because borders will need to have increased contrast in future high-contrast modes
+
+### Contrast requirements
+
+These can also be programmatically validated in tests.
+
+- Color roles as backgrounds (e.g. `bg.primary.default.*`) need at least 4.5:1 contrast with the accompanying `on*` foreground color (e.g. `fg.onPrimary`)
+- Color roles as foregrounds (e.g. `fg.primary`) need at least 4.5:1 contrast with the default background (`bg.default.*`) (Note: see open question about the notify color role)
+- Subtle variants of color roles (e.g. `bg.primary.subtle.*`) need at least 4.5:1 contrast with neutral fg colors (`fg.default`, `fg.subtle`...)
+- Neutral foregrounds need at least 4.5:1 contrast with neutral backgrounds
+- 🚧
+
+### Open questions
+
+- Can we group text and icon colors under `fg`?
+- Is `promo` another color role, or a different thing?
+- In our current light theme, `fg.notify` (e.g. used in notify icons) doesn't have 4.5:1 contrast to `bg.default`. It would be tricky to change this color—can we rely on documentation _not_ to use notify as text color? Or do we need to split icons from text? (`text.notify` could be set as gray, like in the `Input`'s warning message)
+- Foreground colors currently don't have states in our components, except the `Anchor` (becomes darker when hovered) and the tertiary `Button`. Since the Anchor doesn't need it (an underline is enough) and the tertiary button will become underlined, can we remove states for foregrounds? Or do we need to support them?
+- There were open questions about naming for neutral variants. I added `default`, `subtle` and `highlight` in the examples, but heard opinions that e.g. `strong` may make more sense than `highlight` (note: renamings can happen after the v1 of theming)
```

Comment:
Yes I also prefer highlight personally. I'd keep naming changes for later since they don't have to be tied to this initiative, and we currently use `highlight` in e.g. the Body component. FWIW, Primer uses `emphasis` a lot, which I like too